pkg/collector/corechecks/servicediscovery/module/impl_linux.go

```diff
@@ -1188,3 +1209,71 @@ func (s *discovery) getServices(params params) (*model.ServicesResponse, error)
 
 	return response, nil
 }
+
+// getServices processes a list of PIDs and returns service information for each.
+// This is used by the /services endpoint which accepts explicit PID lists and bypasses
+// the port retry logic used by the /check endpoint.
+func (s *discovery) getServices(params params) (*model.ServicesResponse, error) {
+	s.mux.Lock()
+	defer s.mux.Unlock()
+	response := &model.ServicesResponse{
+		Services: make([]model.Service, 0),
+	}
+
+	context := newParsingContext()
+
+	for _, pid := range params.pids {
+		service := s.getServiceWithoutRetry(context, int32(pid))
+		if service == nil {
+			continue
+		}
+		response.Services = append(response.Services, *service)
+	}
+
+	return response, nil
+}
+
+// getServiceWithoutRetry extracts service information for a PID without port retry logic.
+// Unlike getService(), this function immediately returns nil if no ports are found,
+// rather than tracking retry attempts. This is used by the /services endpoint.
+func (s *discovery) getServiceWithoutRetry(context parsingContext, pid int32) *model.Service {
+	if s.shouldIgnorePid(pid) {
+		return nil
+	}
+	if s.shouldIgnoreComm(pid) {
+		return nil
+	}
+
+	ports, err := s.getPorts(context, pid)
+	if err != nil {
+		return nil
+	}
+	if len(ports) == 0 {
+		return nil
+	}
+
+	rss, err := getRSS(pid)
```

Comment:
Why is this endpoint calculating RSS?

Reply:
Good catch. I also forgot to remove the CPU stats check from the tests, so I fixed that too.